aws/resource_aws_cloudwatch_event_archive_test.go

```diff
@@ -0,0 +1,230 @@
+
+	for {
+		output, err := conn.ListArchives(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping CloudWatch Events archive sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving CloudWatch Events archive: %w", err)
+		}
+
+		if len(output.Archives) == 0 {
+			log.Print("[DEBUG] No CloudWatch Events archives to sweep")
+			return nil
+		}
+
+		for _, archive := range output.Archives {
+			name := aws.StringValue(archive.ArchiveName)
+			if name == "default" {
+				continue
+			}
+
+			log.Printf("[INFO] Deleting CloudWatch Events archive (%s)", name)
+			_, err := conn.DeleteArchive(&events.DeleteArchiveInput{
+				ArchiveName: aws.String(name),
+			})
+			if err != nil {
+				return fmt.Errorf("Error deleting CloudWatch Events archive (%s): %w", name, err)
+			}
+		}
+
+		if output.NextToken == nil {
+			break
+		}
+		input.NextToken = output.NextToken
+	}
+
+	return nil
+}
+
+func TestAccAWSCloudWatchArchive_basic(t *testing.T) {
+	var v1 events.DescribeArchiveOutput
+	archiveName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_cloudwatch_event_archive.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "name", archiveName),
+					resource.TestCheckResourceAttr(resourceName, "retention_days", "0"),
+					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "events", fmt.Sprintf("archive/%s", archiveName)),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSCloudWatchArchive_update(t *testing.T) {
+	var v1 events.DescribeArchiveOutput
+	resourceName := "aws_cloudwatch_event_archive.test"
+	archiveName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+				),
+			},
+			{
+				Config: testAccAWSCloudWatchArchiveConfig_updateRetentionAndPattern(archiveName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "retention_days", "7"),
+					testAccCheckResourceAttrEquivalentJSON(resourceName, "event_pattern", "{\"source\":[\"company.team.service\"]}"),
```

Comment:
You should try updating `description` here also to make sure all the updatable attributes work.